### DISCUSSION TITLE
"Un-pop" interactions

### DISCUSSION THREAD

**Discussion by davdroman** [davdroman/swiftui-navigation-transitions#11] [Ideas]
This might be right at the limit of what SwiftUI + Introspect can do without subclassing, but I want to try to introduce ["un-pop" functionality](https://github.com/Enricoza/EZCustomNavigation#right-screen-edge-unpop-disabled-by-default).

Some references in order to explore possible implementations:

- https://stackoverflow.com/questions/31509193/how-do-i-make-an-interactive-uinavigationcontroller-transition-that-can-pop-or-p
- https://stackoverflow.com/questions/57429058/interactive-push-transitions-like-tiktok

**Comment by alexfringes** [comment]
FWIW I'd be very interested in this, espc. including .edgePan and .pan.
The use case I have for it is maybe slightly esoteric, but it's a real one and this library has already greatly helped with surrounding problems. If un-pop were to be an option, that would be amazing.

**Reply by davdroman** [reply]
Good to hear it'd be useful 🙂

I've haven't dug into this yet as I've recently joined the Introspect project as a maintainer and I'm working on a new more stable API for it to hopefully be able to reach v1 status. As soon as that infrastructure is in place I'll revisit this library and tackle this and other goals.

**Reply by alexfringes** [reply]
Yeah I've been eyeing that new branch on Introspect and was actually wondering if it's ready to mess around with. But sounds like a little more time is needed haha. Great work on both libraries.

**Reply by davdroman** [reply]
Great that you're keen to take it for a spin. A proper beta release will come shortly so people can take it for a spin. I'll give you a heads-up when it ready.
